src/main/java/com/eerussianguy/firmalife/items/ItemNutHammer.java

```diff
@@ -53,6 +59,35 @@ public EnumActionResult onItemUse(EntityPlayer player, World worldIn, BlockPos p
                 worldIn.playSound(null, player.getPosition(), SoundEvents.ENTITY_ITEM_BREAK, SoundCategory.PLAYERS, 0.5f, 1.0f);
                 return EnumActionResult.SUCCESS;
             }
+
+            if(worldIn.getBlockState(pos).getBlock() instanceof BlockPlacedItemFlat)
+            {
+                TEPlacedItemFlat tile = (TEPlacedItemFlat) worldIn.getTileEntity(pos);
+                Item item = tile.getStack().getItem();
+
+                if(Constants.RNG.nextInt(2) == 1 && item instanceof IItemFoodTFC) // 50% chance to get a nut
```

Comment:
No reason to check if it's a food item if we're checking the items in the switch statement anyway

Reply:
Without the instanceof check a nullPointerException is thrown whenever the item interacts with something that isn't an instance of IItemFoodTFC